### PR TITLE
No ticket - preload trigger images

### PR DESCRIPTION
**Pull request by EdPutans**
**on slow connections:**
rather than having a slowly loading image, show the trigger only once the browser has loaded it, meaning an instant popup.
This does mean that the trigger occurence **can** get slightly delayed (not seen) on super slow connections, but still makes it a better UX.

### CODE REVIEW

File: src/context/Runners.tsx [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
-import { useInitVisitor } from '../hooks/init/useInitVisitor'
-import { useCollinsBookingComplete } from '../hooks/mab/useCollinsBookingComplete'
-import useButtonCollector from '../hooks/useButtonCollector'
-import { useCollector } from '../hooks/useCollector'
-import useConversions from '../hooks/useConversions'
-import useFormCollector from '../hooks/useFormCollector'
-import useIncompleteTriggers from '../hooks/useIncompleteTriggers'
-import useIntently from '../hooks/useIntently'
-import useWatchers from '../hooks/useWatchers'
+import { useInitSession } from "../hooks/init/useInitSession";
```
Comment (EdPutans) [inline src/context/Runners.tsx:1]: eslint kicking in prematurely🤷‍♀️

File: src/hooks/useImagePreload.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ * // built with Copilot-chat
+ */
+function isValidImageUrl(url: string): boolean {
+  // Regular expression to match common image file extensions
+  const imageExtensions = /\.(jpg|jpeg|png|gif|bmp)$/i;
+
+  // Check if the URL matches the image file extension pattern
+  if (imageExtensions.test(url)) {
+    // Check if the URL starts with a valid protocol (http:// or https://)
+    if (url.startsWith('http://') || url.startsWith('https://')) {
```
Comment (AdenFraser) [inline src/hooks/useImagePreload.ts:17]: This wont work if someone accidently puts a protocol-relative URL in, for instance:
//example.com/image.png

File: src/hooks/useImagePreload.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      return true;
```
Comment (AdenFraser) [inline src/hooks/useImagePreload.ts:18]: Why the nested if? It can exist in one conditional

File: src/hooks/useImagePreload.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+ * Checks if an image is potentially valid based on its URL
+ * // built with Copilot-chat
+ */
+function isValidImageUrl(url: string): boolean {
+  // Regular expression to match common image file extensions
+  const imageExtensions = /\.(jpg|jpeg|png|gif|bmp)$/i;
+
+  // Check if the URL matches the image file extension pattern
+  if (imageExtensions.test(url)) {
+    // Check if the URL starts with a valid protocol (http:// or https://)
```
Comment (AdenFraser) [inline src/hooks/useImagePreload.ts:16]: Does this check actually add any value? What's the side effect if the protocol isn't valid or present?

**Review by AdenFraser — APPROVED** [review]

**Pull request merged**